crates/common/validator/src/sync_committee.rs

```diff
@@ -75,3 +80,32 @@ pub fn compute_subnets_for_sync_committee(
         .map(|index| index as u64 / (SYNC_COMMITTEE_SIZE / SYNC_COMMITTEE_SUBNET_COUNT))
         .collect())
 }
+
+pub fn process_sync_committee_contributions(
+    block: &mut BeaconBlock,
+    contributions: HashSet<SyncCommitteeContribution>,
+) -> anyhow::Result<()> {
+    let mut sync_aggregate = SyncAggregate {
+        sync_committee_bits: BitVector::new(),
+        sync_committee_signature: BLSSignature::infinity(),
+    };
+    let mut signatures: Vec<BLSSignature> = Vec::new();
+    for contribution in contributions {
+        for (index, participated) in contribution.aggregation_bits.iter().enumerate() {
+            if participated {
+                let participant_index = (SYNC_COMMITTEE_SIZE / SYNC_COMMITTEE_SUBNET_COUNT)
+                    * contribution.subcommittee_index
+                    + index as u64;
+                sync_aggregate
+                    .sync_committee_bits
+                    .set(participant_index as usize, true)
+                    .map_err(|err| anyhow::anyhow!("Failed to set sync committee bit: {err:?}"))?;
+            }
+        }
+        signatures.push(contribution.signature);
+    }
+    sync_aggregate.sync_committee_signature =
+        BLSSignature::aggregate(&signatures.iter().collect::<Vec<&BLSSignature>>())?;
+    block.body.sync_aggregate = sync_aggregate;
+    Ok(())
+}
```

Comment:
```suggestion
pub fn process_sync_committee_contributions(
    block: &mut BeaconBlock,
    contributions: HashSet<SyncCommitteeContribution>,
) -> anyhow::Result<()> {
    let mut sync_committee_bits = BitVector<U512>::new();
    let mut signatures = vec![];
    let sync_subcommittee_size = SYNC_COMMITTEE_SIZE / SYNC_COMMITTEE_SUBNET_COUNT;
    
    for contribution in contributions {
        for (index, participated) in contribution.aggregation_bits.iter().enumerate() {
            if participated {
                let participant_index = sync_subcommittee_size
                    * contribution.subcommittee_index
                    + index as u64;
                sync_committee_bits
                    .set(participant_index as usize, true)
                    .map_err(|err| anyhow!("Failed to set sync committee bit: {err:?}"))?;
            }
        }
        signatures.push(contribution.signature);
    }
    
    block.body.sync_aggregate = SyncAggregate {
        sync_committee_bits,
        sync_committee_signature: BLSSignature::aggregate(&signatures.iter().collect::<Vec<&BLSSignature>>())?,
    };
    Ok(())
}
```

This is a bit cleaner